scipy/linalg/flapack_gen.pyf.src

```diff
@@ -990,12 +990,58 @@ subroutine <prefix>geqrf_lwork(m,n,a,tau,work,lwork,info)
     integer intent(in), check(n > 0) :: n
     <ftype> intent(hide) :: a
     <ftype> intent(hide) :: tau
-    integer intent(hide) :: lwork = -1
     <ftype> intent(out) :: work
+    integer intent(hide) :: lwork = -1
     integer intent(out) :: info
 
 end subroutine <prefix>geqrf_lwork
 
+subroutine <prefix>geqrfp(m,n,a,lda,tau,work,lwork,info)
+    ! qr,tau,work,info = geqrfp(a,lwork=3*n,overwrite_a=0)
+    ! DGEQR2P computes a QR factorization of a real M-by-N matrix A:
+    !
+    !    A = Q * ( R ),
+    !            ( 0 )
+    !
+    ! where:
+    !
+    !    Q is a M-by-M orthogonal matrix;
+    !    R is an upper-triangular N-by-N matrix with nonnegative diagonal
+    !    entries;
+    !    0 is a (M-N)-by-N zero matrix, if M > N.
+
+    callstatement (*f2py_func)(&m,&n,a,&lda,tau,work,&lwork,&info)
+    callprotoargument int*,int*,<ctype>*,int*,<ctype>*,<ctype>*,int*,int*
+
+    integer intent(hide), check(m > 0), depend(a) :: m = shape(a, 0)
+    integer intent(hide), check(n > 0), depend(a) :: n = shape(a, 1)
+    <ftype> intent(in,out,copy,out=qr), dimension(m,n) :: a
+    integer intent(hide) :: lda = max(1, m)
```

Comment:
This needs a depend clause but better keep it explicit with `shape(a, 0)` such that f2py doesn't need to resolve any dependency relation.

Reply:
So that should be: ``integer intent(hide), depend(a)  :: lda = max(1,shape(1, 0))``?

Reply:
Yes indeed with `shape(a, 0)`